src/widgets/location/use-locations.tsx

```diff
@@ -0,0 +1,20 @@
+import { useState, useEffect } from "react";
+import { Subscription } from "rxjs";
+import { Location, getLocations } from "./location.resource";
+
+export default function useLocations() {
+  const [locations, setLocations] = useState<Array<Location>>([]);
+  useEffect(() => {
+    let locationSub: Subscription;
```

Comment:
Why not just `const locationSub = ...`? I think the typing would also benefit (no need for the explicit `Subscription`)